addons/stock_calendar/models/resource.py

```diff
@@ -94,21 +79,18 @@ def interval_remove_leaves(self, cr, uid, interval, leave_intervals, context=Non
         :param list leave_intervals: a list of tuples (beginning datetime, ending datetime)
                                     that are intervals to remove from the base interval
         :return list intervals: a list of tuples (begin datetime, end datetime)
-                                that are the remaining valid intervals """
+                                that are the remaining valid intervals
+        """
         if not interval:
             return interval
-        if leave_intervals is None:
-            leave_intervals = []
         intervals = []
-        #leave_intervals = self.interval_clean(leave_intervals) NOT NECESSARY TO CLEAN HERE AS IT WOULD REMOVE GROUP INFO
         current_interval = list(interval)
         for leave in leave_intervals:
             if len(leave) > 2:
                 current_group = False
-                att_obj = self.pool.get("resource.calendar.attendance")
                 if leave[2]:
                     if len(current_interval) > 2:
-                        current_group = current_interval[2] and att_obj.browse(cr, uid, current_interval[2], context=context).group_id.id or False
+                        current_group = self.env['resource.calendar.attendance'].browse(current_interval[2]).group_id.id
```

Comment:
Check should be:
current_interval[2] and self.env['resource.calendar.attendance'].browse(current_interval[2]).group_id.id or False